CON-370, CON-375, CON-376: Further improve the documentation (#2)
rticommunity/rticonnextdds-connector-rust: README.md

# RTI Connector for Rust

RTI® Connext® is a connectivity software framework for integrating data sources
of all types. At its core is the world's leading ultra-high performance,
distributed networking databus.

_RTI Connector_ provides a quick and easy way to write applications that publish

> and subscribe to the RTI Connext DDS databus in Rust and other languages.

Reviewer: ```suggestion
and subscribe to the _RTI Connext_ DDS databus in Rust and other languages.
```

## Documentation

> This crate provides Rust bindings for the _RTI Connector_ API, which allows

Reviewer: ```suggestion
The `rtiddsconnector` crate provides Rust bindings for the _RTI Connector_ API, which allows
```

> easy integration of RTI Connext DDS functionality into Rust applications.

Reviewer: ```suggestion
easy integration of _RTI Connext_ DDS functionality into Rust applications.
```

> easy integration of _RTI Connext_ DDS functionality into Rust applications.

Reviewer: You can find documentation for _Connext_ , _Connector_, and all other RTI products on the following sites:

Author: I'll add this as part of the other manual feedback.

* [RTI Website](https://www.rti.com/)
* [RTI Community](https://community.rti.com/)

> * [API Reference][api-reference]

Reviewer: ```suggestion
* _Connector_ for Rust [API Reference][api-reference]
```

Author: This text would be outside of the link, is that intended?

Reviewer: Nope! It should be inside.

[api-reference]: https://rticommunity.github.io/rticonnextdds-connector-rs/ "RTI Connector for Rust API Reference"

## Quick Start

This crate is intended to be managed through Cargo as a Git repository. To add

> it to your project dependencies, enter:

Reviewer: ```suggestion
`rtiddsconnector` to your project dependencies, enter:
```

```console
cargo add \
    --git https://www.github.com/rticommunity/rticonnextdds-connector-rs \
    --branch <branch-name>
```

`<branch-name>` is the branch you want to use; for example, `master` or a

> specific in-development branch. Omitting `--branch` will default to `master`.

Reviewer: ```suggestion
specific in-development branch. If `--branch` is not defined, `master` is used by default.
```

> Once the crate is available, you can start using it in your rust code. See the

Reviewer: ```suggestion
Once the crate is available, you can start using it in your Rust code. See the
```

quickstart file located at [`snippets/quickstart.rs`](snippets/quickstart.rs)

> for a simple example of how to use the Connector API.

Reviewer: ```suggestion
for a simple example of how to use the _RTI Connector_ API.
```

### Note on Native Libraries

This crate requires the _RTI Connector_ C libraries to be available both during

> build and at runtime.

Reviewer: ```suggestion
the build and at runtime.
```

At build time, the scripts in `build.rs` will attempt to download the required
libraries from the [_RTI Connector_ releases in Github][rti-github-connector].

[rti-github-connector]: https://github.com/rticommunity/rticonnextdds-connector/releases "RTI Connector Github repository"

To tune this behavior, set the environment variable
`RTI_CONNECTOR_VERSION` to the desired version (e.g. `1.4.0`) before building
your application with Cargo. For more information, see
[the HTML documentation][api-reference].

At runtime, ensure that your system's linker can find the
required libraries. This usually involves placing the native libraries
next to the executable or using your system's dynamic library path
environment variable (`LD_LIBRARY_PATH` on Linux, `DYLD_LIBRARY_PATH` on
macOS, or `PATH` on Windows).

## Examples

> The crate includes a `shapes` example compatible with the *RTI Shapes Demo*.

Reviewer: ```suggestion
The crate includes a `shapes` example compatible with *RTI Shapes Demo*.
```

This example can be run as a publisher or as a subscriber. You can read more
about it [in its README](examples/shapes/README.md).

> Additionally, it contains read-only snippets in the `snippets` module that

Reviewer: ```suggestion
Additionally, the example contains read-only snippets in the `snippets` module that
```

> # RTI Connector for Rust

Reviewer: ```suggestion
# RTI Connector for Rust (Experimental)
```

RTI® Connext® is a connectivity software framework for integrating data sources
of all types. At its core is the world's leading ultra-high performance,
distributed networking databus.

_RTI Connector_ provides a quick and easy way to write applications that publish
and subscribe to the _RTI Connext_ DDS databus in Rust and other languages.

> 

Reviewer: Add note callout/blockquote here; not sure how to format that in markdown so plz correct me.

> **IMPORTANT:** _Connector for Rust_ is an experimental product; do not use it in production systems. This release is an evaluation distribution; use it to explore using _RTI Connext_ functionality into Rust applications. For support, you may contact [support@rti.com](mailto:support%40rti.com).

Author: You are using Markdown for writing these comments :)
So, yes, the correct syntax is prepending `>` to a new line to trigger a blockquote.

## Documentation

The `rtiddsconnector` crate provides Rust bindings for the _RTI Connector_ API, which allows
easy integration of _RTI Connext_ DDS functionality into Rust applications.

You can find documentation for Connext, Connector, and all other RTI products
on the following sites:

* [RTI Website](https://www.rti.com/)
* [RTI Community](https://community.rti.com/)

> * [_Connector_ for Rust API Reference][api-reference]

Reviewer: ```suggestion
* [_Connector for Rust_ API Reference][api-reference]
```

`rtiddsconnector` to your project dependencies, enter:

```console
cargo add \
    --git https://www.github.com/rticommunity/rticonnextdds-connector-rs \
    --branch <branch-name>
```

`<branch-name>` is the branch you want to use; for example, `master` or a
specific in-development branch. If `--branch` is not defined, `master` is used by default.

Once the crate is available, you can start using it in your Rust code. See the
quickstart file located at [`snippets/quickstart.rs`](snippets/quickstart.rs)
for a simple example of how to use the _RTI Connector_ API.

### Note on Native Libraries

This crate requires the _RTI Connector_ C libraries to be available both during
the build and at runtime.

At build time, the scripts in `build.rs` will attempt to download the required
libraries from the [_RTI Connector_ releases in Github][rti-github-connector].

[rti-github-connector]: https://github.com/rticommunity/rticonnextdds-connector/releases "RTI Connector Github repository"

To tune this behavior, set the environment variable
`RTI_CONNECTOR_VERSION` to the desired version (e.g. `1.4.0`) before building
your application with Cargo. For more information, see

> [the HTML documentation][api-reference].

Reviewer: ```suggestion
[the _Connector for Rust_ API reference][api-reference].
```

At runtime, ensure that your system's linker can find the
required libraries. This usually involves placing the native libraries
next to the executable or using your system's dynamic library path
environment variable (`LD_LIBRARY_PATH` on Linux, `DYLD_LIBRARY_PATH` on
macOS, or `PATH` on Windows).

## Examples

The crate includes a `shapes` example compatible with *RTI Shapes Demo*.
This example can be run as a publisher or as a subscriber. You can read more
about it [in its README](examples/shapes/README.md).

Additionally, the example contains read-only snippets in the `snippets` module that
demonstrate various features of the Connector API, and are used in the
documentation.

> 

Reviewer: Recommended adding this language as first paragraph of the License section, based on legal's note:

RTI Connector for Rust is part of the Connext Professional Package. If you have a valid license for the RTI Connext Professional Package, such license shall govern your use of RTI Connector for Rust. All other use of this software shall be governed solely by the terms of RTI’s Software License for Non-Commercial Use #4040, included at the top level of the this repository.

# RTI Connector for Rust (Experimental)

RTI® Connext® is a connectivity software framework for integrating data sources
of all types. At its core is the world's leading ultra-high performance,
distributed networking databus.

_RTI Connector_ provides a quick and easy way to write applications that publish
and subscribe to the _RTI Connext_ DDS databus in Rust and other languages.

_RTI Connector for Rust_ is an experimental crate that provides Rust bindings
for the _RTI Connector_ API. It does so by exposing a safe and idiomatic Rust
interface over the underlying C API, in a crate called `rtiddsconnector`.

> **IMPORTANT:**
> _Connector for Rust_ is an experimental product; do not use it in production
> systems. This release is an evaluation distribution; use it to explore using

> > _RTI Connext_ functionality into Rust applications. For support, you may

Reviewer: I learned support@rti.com is not the proper contact for experimental products. Let's replace the last sentence in the callout with the following:
" As an experimental RTI product, we only offer support through the [RTI Community Forum](https://community.rti.com/forum), backed by RTI engineers who will answer your questions.

## Quick Start

This crate is intended to be managed through Cargo as a Git repository. To add
`rtiddsconnector` to your project dependencies, enter:

```console
cargo add \
    --git https://www.github.com/rticommunity/rticonnextdds-connector-rs \
    --branch <branch-name>
```

`<branch-name>` is the branch you want to use; for example, `master` or a
specific in-development branch. If `--branch` is not defined, `master` is used
by default.

Once the crate is available, you can start using it in your Rust code. See the
quickstart file located at [`snippets/quickstart.rs`](snippets/quickstart.rs)
for a simple example of how to use the _RTI Connector_ API.

### Note on Native Libraries

This crate requires the _RTI Connector_ C libraries to be available both during
the build and at runtime.

At build time, the scripts in `build.rs` will attempt to download the required
libraries from the [_RTI Connector_ releases in Github][rti-github-connector].

[rti-github-connector]: https://github.com/rticommunity/rticonnextdds-connector/releases "RTI Connector Github repository"

To tune this behavior, set the environment variable

> `RTI_CONNECTOR_VERSION` to the desired version (e.g. `1.4.0`) before building

Reviewer: ```suggestion
`RTI_CONNECTOR_VERSION` to the desired version (e.g., `1.4.0`) before building
```

your application with Cargo. For more information, see
[the _Connector for Rust_ API reference][api-reference].

At runtime, ensure that your system's linker can find the
required libraries. This usually involves placing the native libraries
next to the executable or using your system's dynamic library path
environment variable (`LD_LIBRARY_PATH` on Linux, `DYLD_LIBRARY_PATH` on
macOS, or `PATH` on Windows).

## Examples

> The crate includes a `shapes` example compatible with *RTI Shapes Demo*.

Reviewer: ```suggestion
The crate includes a `shapes` example compatible with *[RTI Shapes Demo](https://community.rti.com/static/documentation/connext-dds/current/doc/manuals/connext_dds_professional/tools/shapes_demo/shapes_demo/ShapesTitle.htm)*.
```

This example can be run as a publisher or as a subscriber. You can read more
about it [in its README](examples/shapes/README.md).

Additionally, the example contains read-only snippets in the `snippets` module
that demonstrate various features of the _Connector_ API, and are used in the
documentation.

## License

_RTI Connector_ for Rust is part of the RTI Connext Professional Package.
If you have a valid license for the RTI Connext Professional Package,
such license shall govern your use of _RTI Connector_ for Rust.
All other use of this software shall be governed solely by the terms of
RTI’s Software License for Non-Commercial Use \#4040, included at the

> [top level of the this repository](LICENSE.pdf).

Reviewer: does legal need to review this section?

```suggestion
[top level of this repository](LICENSE.pdf).
```

The `rtiddsconnector` crate provides Rust bindings for the _RTI Connector_ API, which allows
easy integration of _RTI Connext_ DDS functionality into Rust applications.

> You can find documentation for Connext , Connector, and all other RTI products

Reviewer: ```suggestion
You can find documentation for Connext, Connector, and all other RTI products
```
